### ISSUE TITLE
Rule LANG - The byte order mark (BOM) is detected?

### ISSUE THREAD

**Issue by anibalsanchez** [joomla-extensions/jedchecker#123]
I'm testing a new development version. 

Processing an extension, there's an error message: **The byte order mark (BOM) is detected**

This is a lack of knowledge from my side. What is the problem with the BOM? 

CC @dryabov

**Comment by dryabov** [comment]
The problem is that BOM is a part of the following translation key, and as a result it is not processed by Joomla. For example, let one have the following INI file:
```ini
<BOM>COM_MYEXT="My Ext"
COM_MYEXT_SMTH="..."
```
(where `<BOM>` is the 3-byte UTF-8 BOM sequence) and the following PHP file:
```php
<?php echo JText::_('COM_MYEXT');
```
In this case Joomla is not able to find translation for `COM_MYEXT`. To bypass this error, many translation files have a comment in the first line:
```ini
<BOM>; Translations for My Ext
COM_MYEXT="My Ext"
COM_MYEXT_SMTH="..."
```
and that's why the Language rule displays error message in the former case, and just a warning in the latter one.

**Comment by anibalsanchez** [comment]
Thanks for the clarification. 

Is this documented somewhere? It would be great if we can include a link as a reference. I'm sure that people are going to start asking what is it.

**Comment by dryabov** [comment]
https://docs.joomla.org/Language_Guidelines_for_3rd_Party_Extensions:
> All .ini language files need to be saved as **UTF-8 no BOM** for the proper display of characters which are not in the ASCII range [a..z] and [A..Z].

https://docs.joomla.org/J3.x:Making_a_Language_Pack_for_Joomla:
> ALL FILES MUST BE SAVED AS **UTF8 NO BOM** (for non-ascii glyphs as in accented letters, etc.)

But none explains why it's necessary.

**Comment by anibalsanchez** [comment]
Great. Case closed.
